rpcs3/Emu/Cell/PPUModule.cpp

```diff
@@ -825,6 +825,27 @@ std::shared_ptr<lv2_prx> ppu_load_prx(const ppu_prx_object& elf, const std::stri
 	u32 end = 0;
 	u32 toc = 0;
 
+	u32 prx_mem_size = 0;
+
+	for (const auto& prog : elf.progs)
+	{
+		if (prog.p_type == 0x1u /* LOAD */ && prog.p_memsz)
+		{
+			prx_mem_size += utils::align(static_cast<u32>(prog.p_memsz), 0x10000);
```

Comment:
I think it's a bit more complicated... And you can use page alignment 4096 for first segment, and prog.p_align for further segments. In theory.

Reply:
I can even assert it with ENOEXEC if its not equal to exactly 0x10000 im nearly sure thats what happens on realfw from previous disasm sessions in the past.
Cell PPU ABI suggests it as well.
vm::main supports only 64k pages.

Reply:
You are squeezing segments into a single allocation. I'm not sure about first seg, but second+ can easily be squeezed and perhaps you should take into account their alignment. Someone would also ask to extend executable one for a code cave.

Reply:
Remember that each segment is unique and has its own flags such as PPU/SPU r/w/exec attrubutes, hance 64k is the min alignment fir even on realhw.

Reply:
I mean, that's how sys_memory_get_page_attribute works so its for that, if it was okay to lower memory consumption without introducing future bugs I'd done it.